tests/base/error.cpp

```diff
@@ -0,0 +1,57 @@
+//  This file is part of Empirical, https://github.com/devosoft/Empirical
+//  Copyright (C) Michigan State University, 2016-2017.
+//  Released under the MIT Software license; see doc/LICENSE
+
+#define CATCH_CONFIG_MAIN
+#undef NDEBUG
+#define TDEBUG 1
+
+#include "third-party/Catch/single_include/catch2/catch.hpp"
+
+#include "emp/base/error.hpp"
+
+#include <sstream>
+#include <iostream>
+
+TEST_CASE("Test error", "[base]")
+{
+  // Error should not be thrown by default
+  REQUIRE_FALSE(emp::error_thrown);
+  // Throw an error and try to detect it! 
+  std::string passed_error_string = "This is an error!";
+  // Don't separate the next two lines or else the line number test will fail!
+  size_t expected_error_line = __LINE__ + 1;
+  emp_error(passed_error_string);
+  REQUIRE(emp::error_thrown);
+
+  // Did we capture the correct filename and line number? 
+  // Trim the filename in case the test is run from elsewhere
+  std::string caught_filename = emp::error_info.filename;
+  std::string expected_filename = "error.cpp";
+	REQUIRE(caught_filename.substr(caught_filename.size()-expected_filename.size()) == expected_filename);
```

Comment:
Seems like a good opportunity for Catch marchers https://github.com/catchorg/Catch2/blob/devel/docs/matchers.md#using-matchers

also, indentation?